pkg/virt-launcher/virtwrap/converter/vcpu/vcpu.go

```diff
@@ -453,19 +470,12 @@ func AdjustDomainForTopologyAndCPUSet(domain *api.Domain, vmi *v12.VirtualMachin
 		}
 	}
 
-	var emulatorThreads []uint32
 	var emulatorThreadsCPUSet string
 	if vmi.Spec.Domain.CPU.IsolateEmulatorThread {
-		availableThread, err := cpuPool.FitThread()
-		if err != nil {
-			e := fmt.Errorf("no CPU allocated for the emulation thread: %v", err)
-			log.Log.Reason(e).Error("failed to format emulation thread pin")
-			return e
+		if emulatorThreadsCPUSet, err = FormatDomainEmulatorThreadPin(domain, cpuPool); err != nil {
```

Comment:
suggestion: Move appendDomainEmulatorThreadPin outside of this function to keep the logic separate from the execution